### PR TITLE
add optional lifecyclepolicy option

### PR DESCRIPTION
**Pull request by bennycao**
add extra plugin option to pass in the path to the repos lifecycle policy file so a custom policy can be used
rename existing policy option names to allow for the above

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,17 +10,19 @@ A [Buildkite plugin](https://buildkite.com/docs/agent/v3/plugins) to create and
...
+        lifecyclepolicy: 'path/to/lifecycle-policy.json'
     command:
       - echo hi
 ```
 
 Params:
 
 - name (required) - name of the ECR.
-- policy (optional) - path in local repository to the policy file.
+- permissionspolicy (optional) - path in local repository to the permissions policy file.
```
Comment (bennycao) [inline README.md]: don't necessary like the names 🈂️

File: plugin.yaml [hunk 2]
```diff
@@ -7,6 +7,8 @@ configuration:
   properties:
     name:
       type: string
-    policy:
+    permissionspolicy:
```
Comment (72636c) [inline plugin.yaml]: Can we kebab-case the options?

Comment (bennycao) [reply plugin.yaml]: does that work ?

Comment (72636c) [reply plugin.yaml]: - https://github.com/buildkite-plugins/ecr-buildkite-plugin/blob/df34ad30e592c44e450f02e14366f75bf9beb237/plugin.yml#L17
- https://github.com/buildkite-plugins/ecr-buildkite-plugin/blob/df34ad30e592c44e450f02e14366f75bf9beb237/hooks/pre-command#L69

File: hooks/pre-command [hunk 3]
```diff
@@ -26,19 +26,28 @@ upsert_ecr() {
...
 		echo 'Setting ECR permissions'
     aws ecr set-repository-policy \
       --repository-name "${repository_name}" \
-      --policy-text "file://${policy_file}"
+      --policy-text "file://${permissions_policy_file}"
   fi
 
 	echo 'Setting ECR Lifecycle policy'
-  aws ecr put-lifecycle-policy \
+  lifecycle_policy_file="${BUILDKITE_PLUGIN_CREATE_ECR_LIFECYCLE_POLICY:-''}"
```
Comment (runk) [inline hooks/pre-command]: You can use `$basedir/policies/default-lifecycle-policy.json` as default value on L39 and don't do if/else

File: hooks/pre-command [hunk 4]
```diff
@@ -26,19 +26,28 @@ upsert_ecr() {
     aws ecr create-repository --repository-name "${repository_name}"
   fi
 
-	policy_file="${BUILDKITE_PLUGIN_CREATE_ECR_POLICY:-''}"
-	echo "Policy file: ${policy_file}"
-  if [ "$policy_file" != "" ]; then
+	permissions_policy_file="${BUILDKITE_PLUGIN_CREATE_ECR_PERMISSIONS_POLICY:-''}"
+	echo "Policy file: ${permissions_policy_file}"
+  if [ "$permissions_policy_file" != "" ]; then
 		echo 'Setting ECR permissions'
```
Comment (72636c) [inline hooks/pre-command]: Should we call it a _repository policy_ to align with AWS?

**Review by 72636c — APPROVED** [review]

File: hooks/pre-command [hunk 5]
```diff
@@ -26,19 +26,20 @@ upsert_ecr() {
     aws ecr create-repository --repository-name "${repository_name}"
   fi
 
-	policy_file="${BUILDKITE_PLUGIN_CREATE_ECR_POLICY:-''}"
-	echo "Policy file: ${policy_file}"
-  if [ "$policy_file" != "" ]; then
+	permissions_policy_file="${BUILDKITE_PLUGIN_CREATE_ECR_PERMISSIONS_POLICY:-''}"
```
Comment (runk) [inline hooks/pre-command]: Hmm. Indentation is all over the place, perhaps I screwed it up. 😢 https://github.com/seek-oss/create-ecr-buildkite-plugin/blob/f616f3106647738799d55d535b8b86f0ec6e4496/hooks/pre-command#L29

**Pull request merged**